elrond-wasm/src/api/send_api.rs

```diff
@@ -277,6 +295,17 @@ where
 		self.call_local_esdt_built_in_function(gas, b"ESDTNFTBurn", &arg_buffer);
 	}
 
+	/// Burns ESDT tokens. Handles any type of ESDT.
+	fn burn_esdt(&self, token: &TokenIdentifier, nonce: u64, amount: &BigUint, gas: u64) {
```

Comment:
I would rather call this method burn_tokens. You are handling both ESDTs and NFTs in the same method